hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpGetCommandProcessor.java

```diff
@@ -337,16 +329,26 @@ private JsonObject toJson(CPSession cpSession) {
      */
     private void handleCluster(HttpGetCommand command) {
         Node node = textCommandService.getNode();
-        StringBuilder res = new StringBuilder(node.getClusterService().getMemberListString());
-        res.append("\n");
         NetworkingService ns = node.getNetworkingService();
         EndpointManager cem = ns.getEndpointManager(CLIENT);
         AggregateEndpointManager aem = ns.getAggregateEndpointManager();
-        res.append("ConnectionCount: ").append(cem.getActiveConnections().size());
-        res.append("\n");
-        res.append("AllConnectionCount: ").append(aem.getActiveConnections().size());
-        res.append("\n");
-        command.setResponse(null, stringToBytes(res.toString()));
+        ClusterServiceImpl clusterService = node.getClusterService();
+        JsonArray membersArray = new JsonArray();
+        clusterService.getMembers()
+                      .stream()
+                      .map(m -> new JsonObject()
+                              .add("address", m.getAddress().toString())
+                              .add("liteMember", m.isLiteMember())
+                              .add("localMember", m.localMember())
+                              .add("uuid", m.getUuid().toString())
+                              .add("memberVersion", m.getVersion().toString()))
+                      .forEach(membersArray::add);
+        JsonObject response = new JsonObject()
+                .add("memberList", clusterService.getMemberListString())
```

Comment:
Unless there is some reason to keep the old string version, I'd consider removing `memberList` at all.

Reply:
I guess since all of the data is accessible through `members` (except for member list version, which is not a very easy thing to reason about and a bit internal), I'll remove it 👍